src/core/mesh/qgstriangularmesh.cpp

```diff
@@ -193,11 +199,54 @@ void QgsTriangularMesh::update( QgsMesh *nativeMesh, QgsRenderContext *context )
   // CALCULATE SPATIAL INDEX
   mSpatialIndex = QgsMeshSpatialIndex( mTriangularMesh );
 
-  // SET ALL TRIANGLE CCW
+  // SET ALL TRIANGLE CCW AND COMPUTE AVERAGE SIZE
+  finalizeTriangles();
+
+  QgsEventTracing::addEvent( QgsEventTracing::End, "Triangular mesh", "update" );
+
+  return true;
+}
+
+void QgsTriangularMesh::finalizeTriangles()
+{
+  QgsEventTracing::addEvent( QgsEventTracing::Begin, "Triangular mesh", "finalize" );
+  mAverageTriangleSize = 0;
   for ( int i = 0; i < mTriangularMesh.faceCount(); ++i )
   {
-    setTrianglesCounterClockWise( mTriangularMesh.faces[i] );
```

Comment:
If the function is not used anymore, we can remove it to avoid code duplication?